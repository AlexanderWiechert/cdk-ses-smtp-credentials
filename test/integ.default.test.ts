import { Template } from "aws-cdk-lib/assertions";
import { stack } from "../src/integ.default";

describe("integ.default", () => {
  it("Should match snapshot", () => {
    // When
    const t = Template.fromStack(stack);
    expect(t).toMatchSnapshot();
  });
});
