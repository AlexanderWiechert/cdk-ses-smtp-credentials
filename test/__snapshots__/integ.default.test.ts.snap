// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integ.default Should match snapshot 1`] = `
Object {
  "Mappings": Object {
    "LatestNodeRuntimeMap": Object {
      "af-south-1": Object {
        "value": "nodejs20.x",
      },
      "ap-east-1": Object {
        "value": "nodejs20.x",
      },
      "ap-northeast-1": Object {
        "value": "nodejs20.x",
      },
      "ap-northeast-2": Object {
        "value": "nodejs20.x",
      },
      "ap-northeast-3": Object {
        "value": "nodejs20.x",
      },
      "ap-south-1": Object {
        "value": "nodejs20.x",
      },
      "ap-south-2": Object {
        "value": "nodejs20.x",
      },
      "ap-southeast-1": Object {
        "value": "nodejs20.x",
      },
      "ap-southeast-2": Object {
        "value": "nodejs20.x",
      },
      "ap-southeast-3": Object {
        "value": "nodejs20.x",
      },
      "ap-southeast-4": Object {
        "value": "nodejs20.x",
      },
      "ap-southeast-5": Object {
        "value": "nodejs20.x",
      },
      "ap-southeast-7": Object {
        "value": "nodejs20.x",
      },
      "ca-central-1": Object {
        "value": "nodejs20.x",
      },
      "ca-west-1": Object {
        "value": "nodejs20.x",
      },
      "cn-north-1": Object {
        "value": "nodejs18.x",
      },
      "cn-northwest-1": Object {
        "value": "nodejs18.x",
      },
      "eu-central-1": Object {
        "value": "nodejs20.x",
      },
      "eu-central-2": Object {
        "value": "nodejs20.x",
      },
      "eu-isoe-west-1": Object {
        "value": "nodejs18.x",
      },
      "eu-north-1": Object {
        "value": "nodejs20.x",
      },
      "eu-south-1": Object {
        "value": "nodejs20.x",
      },
      "eu-south-2": Object {
        "value": "nodejs20.x",
      },
      "eu-west-1": Object {
        "value": "nodejs20.x",
      },
      "eu-west-2": Object {
        "value": "nodejs20.x",
      },
      "eu-west-3": Object {
        "value": "nodejs20.x",
      },
      "il-central-1": Object {
        "value": "nodejs20.x",
      },
      "me-central-1": Object {
        "value": "nodejs20.x",
      },
      "me-south-1": Object {
        "value": "nodejs20.x",
      },
      "mx-central-1": Object {
        "value": "nodejs20.x",
      },
      "sa-east-1": Object {
        "value": "nodejs20.x",
      },
      "us-east-1": Object {
        "value": "nodejs20.x",
      },
      "us-east-2": Object {
        "value": "nodejs20.x",
      },
      "us-gov-east-1": Object {
        "value": "nodejs18.x",
      },
      "us-gov-west-1": Object {
        "value": "nodejs18.x",
      },
      "us-iso-east-1": Object {
        "value": "nodejs18.x",
      },
      "us-iso-west-1": Object {
        "value": "nodejs18.x",
      },
      "us-isob-east-1": Object {
        "value": "nodejs18.x",
      },
      "us-west-1": Object {
        "value": "nodejs20.x",
      },
      "us-west-2": Object {
        "value": "nodejs20.x",
      },
    },
  },
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "CredentialsLambda426A03FE": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "CredentialsSecret40F8BB41",
        "User00B015A1",
      ],
      "Properties": Object {
        "SecretId": Object {
          "Ref": "CredentialsSecret40F8BB41",
        },
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialsproviderframeworkonEvent1376F76C",
            "Arn",
          ],
        },
        "UserName": Object {
          "Ref": "User00B015A1",
        },
      },
      "Type": "AWS::CloudFormation::CustomResource",
      "UpdateReplacePolicy": "Delete",
    },
    "CredentialsPolicy6B788293": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "ses:SendRawEmail",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CredentialsPolicy6B788293",
        "Users": Array [
          Object {
            "Ref": "User00B015A1",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CredentialsSecret40F8BB41": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "Description": Object {
          "Fn::Join": Array [
            "",
            Array [
              "SES Smtp Credentials (username, password) for ",
              Object {
                "Ref": "User00B015A1",
              },
            ],
          ],
        },
        "GenerateSecretString": Object {},
      },
      "Type": "AWS::SecretsManager::Secret",
      "UpdateReplacePolicy": "Delete",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A": Object {
      "DependsOn": Array [
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "2819175352ad1ce0dae768e83fc328fb70fb5f10b4a8ff0ccbcb791f02b0716d.zip",
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
            "Arn",
          ],
        },
        "Runtime": Object {
          "Fn::FindInMap": Array [
            "LatestNodeRuntimeMap",
            Object {
              "Ref": "AWS::Region",
            },
            "value",
          ],
        },
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:PutRetentionPolicy",
                "logs:DeleteRetentionPolicy",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
        "Roles": Array [
          Object {
            "Ref": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "User00B015A1": Object {
      "Properties": Object {
        "UserName": "SesTestUser",
      },
      "Type": "AWS::IAM::User",
    },
    "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialshandler320CA55D": Object {
      "DependsOn": Array [
        "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialshandlerServiceRoleDefaultPolicyA3895E80",
        "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialshandlerServiceRole18BEBF78",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "681d284a27fcbbccf41227929a9c13895a5859749b289f961ca68f3b8d47d137.zip",
        },
        "Description": "src/provider/credentials-handler.lambda.ts",
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialshandlerServiceRole18BEBF78",
            "Arn",
          ],
        },
        "Runtime": "nodejs22.x",
        "Timeout": 60,
      },
      "Type": "AWS::Lambda::Function",
    },
    "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialshandlerLogRetention5C8B4A91": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialshandler320CA55D",
              },
            ],
          ],
        },
        "RetentionInDays": 30,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialshandlerServiceRole18BEBF78": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialshandlerServiceRoleDefaultPolicyA3895E80": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "iam:CreateAccessKey",
                "iam:DeleteAccessKey",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "User00B015A1",
                    "Arn",
                  ],
                },
              ],
            },
            Object {
              "Action": "secretsmanager:PutSecretValue",
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Ref": "CredentialsSecret40F8BB41",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialshandlerServiceRoleDefaultPolicyA3895E80",
        "Roles": Array [
          Object {
            "Ref": "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialshandlerServiceRole18BEBF78",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialsproviderframeworkonEvent1376F76C": Object {
      "DependsOn": Array [
        "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialsproviderframeworkonEventServiceRoleDefaultPolicyB231CF26",
        "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialsproviderframeworkonEventServiceRoleFE45B538",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "8ba5ee0003de28bf8e4a4fc7e1821c881abd3a80657825533ed1c301bec7e74a.zip",
        },
        "Description": "AWS CDK resource provider framework - onEvent (SesSmtpCredentialsIntegrationTest/cdk-ses-smtp-credentials.CredentialsProvider/ses-smtp-credentials-provider)",
        "Environment": Object {
          "Variables": Object {
            "USER_ON_EVENT_FUNCTION_ARN": Object {
              "Fn::GetAtt": Array [
                "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialshandler320CA55D",
                "Arn",
              ],
            },
          },
        },
        "Handler": "framework.onEvent",
        "Role": Object {
          "Fn::GetAtt": Array [
            "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialsproviderframeworkonEventServiceRoleFE45B538",
            "Arn",
          ],
        },
        "Runtime": Object {
          "Fn::FindInMap": Array [
            "LatestNodeRuntimeMap",
            Object {
              "Ref": "AWS::Region",
            },
            "value",
          ],
        },
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialsproviderframeworkonEventLogRetentionCD61E4D7": Object {
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialsproviderframeworkonEvent1376F76C",
              },
            ],
          ],
        },
        "RetentionInDays": 1,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
            "Arn",
          ],
        },
      },
      "Type": "Custom::LogRetention",
    },
    "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialsproviderframeworkonEventServiceRoleDefaultPolicyB231CF26": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialshandler320CA55D",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialshandler320CA55D",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialsproviderframeworkonEventServiceRoleDefaultPolicyB231CF26",
        "Roles": Array [
          Object {
            "Ref": "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialsproviderframeworkonEventServiceRoleFE45B538",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "cdksessmtpcredentialsCredentialsProvidersessmtpcredentialsproviderframeworkonEventServiceRoleFE45B538": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
